### PR TITLE
removes deprecated description and cause for std::error::Error

### LINKED ISSUE #7202: Incorrect Error Message when Private Habitat Package Loaded without Authorization
Trying to load a private Habitat package without exporting my **HAB_AUTH_TOKEN** I get the following:

```
# hab svc load <pkg-ident>
» Installing <pkg-ident>
↓ Downloading <pkg-ident>
↓ Downloading <pkg-ident>
↓ Downloading <pkg-ident>
↓ Downloading <pkg-ident>
↓ Downloading <pkg-ident>
↓ Downloading <pkg-ident>
✗✗✗
✗✗✗ [Err: 0] We tried 5 times but could not download <pkg-ident>. Last error was: description() is deprecated; use Display
✗✗✗
```
Once I exported the token, the command worked.

This should probably be more descriptive/indicative of the real error.

### PR DESCRIPTION
**Pull request by jeremymv2**
closes https://github.com/habitat-sh/habitat/issues/7202

Signed-off-by: Jeremy J. Miller <jm@chef.io>

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -208,26 +208,11 @@ where
     E: StdError,
 {
     fn fmt(&self, formatter: &mut Formatter) -> Result<(), FmtError> {
-        write!(formatter, "{}", self.description())
-    }
-}
-
-impl<E> StdError for Error<E>
```
Comment (davidMcneil) [inline src/lib.rs:215]: I think we want to still `impl` `std::error::Error`, but now its just `impl<E: StdError> for Error<E> {}`

Comment (davidMcneil) [reply src/lib.rs:215]: If you wanted to you could also implement `source`. So the whole thing would be something like:

```rust
impl<E: StdError> for Error<E> {
    fn source(&self) -> Option<&(dyn Error + 'static)> {
        if let Self::Operation { error, .. } = self {
            Some(error)
        } else {
            None
        }
    }
}
```

**Review by davidMcneil — CHANGES_REQUESTED** [review]
Nice find!

File: src/lib.rs (hunk 1 above)
Comment (jeremymv2) [reply src/lib.rs:215]: @davidMcneil good call! I had to modify a bit to keep the compiler happy with a lifetime in https://github.com/habitat-sh/retry/pull/1/commits/e7bdff77fe42eacb3f04fdfa949fd99e878a62cb

**Review by davidMcneil — APPROVED** [review]

**Pull request merged**